File: coding/04th July 2025/Robot Return to Origin.java
//Robot Return to Origin
class Solution 
{
    public boolean judgeCircle(String moves) 
    {
        int i = 0, j = 0;
        for(char ch : moves.toCharArray())
        {
            switch(ch)
            {
                case 'R':
                    j++;
                    break;
                case 'L':
                    j--;
                    break;
                case 'U':
                    i--;
                    break;
                case 'D':
                    i++;
                    break;
            }
        }
        return i == 0 && j == 0;
    }
}

File: Coding in Python/05th April 2026/Robot Return to Origin.py
#Robot Return to Origin
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        i = 0
        j = 0
        for ch in moves:
            if ch == 'R':
                j += 1
            elif ch == 'L':
                j -= 1
            elif ch == 'U':
                i -= 1
            elif ch == 'D':
                i += 1
        return i == 0 and j == 0
